k3k-kubelet/controller/webhook/pod.go

```diff
@@ -63,10 +67,27 @@ func (w *webhookHandler) Default(ctx context.Context, obj runtime.Object) error
 	if !ok {
 		return fmt.Errorf("invalid request: object was type %t not cluster", obj)
 	}
-	w.logger.Infow("recieved request", "Pod", pod.Name, "Namespace", pod.Namespace)
+	w.logger.Infow("mutator webhook request", "Pod", pod.Name, "Namespace", pod.Namespace)
 	if pod.Spec.NodeName == "" {
 		pod.Spec.NodeName = w.nodeName
 	}
+	// look for status.* fields in the env
+	if pod.Annotations == nil {
+		pod.Annotations = make(map[string]string)
+	}
+	for i, container := range pod.Spec.Containers {
+		for j, env := range container.Env {
+			if env.ValueFrom != nil {
+				if env.ValueFrom.FieldRef != nil {
+					if strings.Contains(env.ValueFrom.FieldRef.FieldPath, "status.") {
+						pod.Annotations[FieldpathField+"_"+strconv.Itoa(i)+"_"+env.Name] = env.ValueFrom.FieldRef.FieldPath
+						pod.Spec.Containers[i].Env = removeEnv(pod.Spec.Containers[i].Env, j)
+					}
+				}
+			}
+		}
+	}
```

Comment:
When possible I try to avoid too many nested ifs, what about a small rewrite like:

```suggestion
	// look for status.* fields in the env
	if pod.Annotations == nil {
		pod.Annotations = make(map[string]string)
	}
	for i, container := range pod.Spec.Containers {
		for j, env := range container.Env {
			if env.ValueFrom == nil || env.ValueFrom.FieldRef == nil {
				continue
			}
			
			fieldPath := env.ValueFrom.FieldRef.FieldPath
			if strings.Contains(fieldPath, "status.") {
				annotationKey := fmt.Sprintf("%s_%d_%s", FieldpathField, i, env.Name)
				pod.Annotations[annotationKey] = fieldPath
				pod.Spec.Containers[i].Env = removeEnv(pod.Spec.Containers[i].Env, j)
			}
		}
	}
```

Also because of the strconv and all the params I find cleaner to use a `fmt.Sprintf`.

I would also move this logic in a separate func, with a specific name and a small comment on why it's needed. Something like `cleanupFieldRef`.